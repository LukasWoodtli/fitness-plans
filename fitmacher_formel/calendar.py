from datetime import datetime

from icalendar import Event, Calendar


class CalendarEvent:
    def __init__(self, summary, date, description=""):
        self.summary = summary
        self.date = date
        self.description = description

    def get_event(self):
        event = Event()
        event.add('summary', self.summary)
        event.add('dtstart', self.date)
        event.add('dtend', self.date)
        event.add('dtstamp', datetime.now())
        event.add('description', self.description)
        return event

    def add_to_cal(self, cal):
        cal.add_component(self.get_event())


def add_to_calender(workout_events):
    cal = Calendar()

    cal.add('prodid', "Fitness")
    cal.add('version', '2.0')
    [event.add_to_cal(cal) for event in workout_events]

    return cal