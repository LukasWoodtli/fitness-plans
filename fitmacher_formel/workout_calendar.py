from datetime import datetime

from icalendar import Event
from icalendar import Calendar


class CalendarEvent:
    def __init__(self, summary, date, description=""):
        self.summary = summary
        self.date = date
        self.description = description

    def get_event_as_ical(self):
        event = Event()
        event.add('summary', self.summary)
        event.add('dtstart', self.date)
        event.add('dtend', self.date)
        event.add('dtstamp', datetime.now())
        event.add('description', self.description)
        return event


class WorkoutCalendar:
    def __init__(self, workouts):
        super().__init__()
        self.cal = Calendar()
        self._add_to_calender(workouts)

    def _add_to_calender(self, workout_events):
        self.cal.add('prodid', "Fitness")
        self.cal.add('version', '2.0')
        for event in workout_events:
            self.cal.add_component(event.get_event_as_ical())

    def get_all_as_ical(self):
        return self.cal.to_ical().decode('utf-8')