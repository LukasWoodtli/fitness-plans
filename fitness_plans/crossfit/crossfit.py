#!/usr/bin/env python3
# -*- coding: utf-8 -*-
import base64
import datetime
import json
import os
import re
from datetime import date, timedelta

from fitness_plans.lib.abstract_fitness_plan import FitnessPlan
from fitness_plans.lib.util import read_base64_file
from fitness_plans.lib.workout_calendar import CalendarEvent

SCRIPT_DIR = os.path.dirname(os.path.realpath(__file__))


class CrossFit(FitnessPlan):

    def __init__(self, workout_start_date=date.today() + datetime.timedelta(days=1), first_workout=1, output_dir=None):
        super().__init__()
        self.calendar_name = "Cross Fit"
        self.workout_start = workout_start_date
        self.first_workout = first_workout
        self.output_dir = output_dir

    def get_input(self):
        ret_encoded = b'ewogICIxIjogIkVNT1RNIGZcdTAwZmNyIDUgTWludXRlbjogMTUgQWlyIFNxdWF0c1xuQXVmIFplaXQ6IDEwLi4uM' \
              b'SBQdXNoLVVwIHVuZCBTaXQtVXAiLAogICI0IjogIkVNT1RNIGZcdTAwZmNyIDUgTWludXRlbjpcbjE1IFJ1c3NpYW4' \
              b'gU3dpbmdzXG5BdWYgWmVpdDogMjUwIFJvcGUgSnVtcHMiLAogICI2IjogIjEwLi4uMSBHb2JsZXQgU3F1YXRzICYgU' \
              b'HVzaC1VcHNcbkludGVydmFsbGUgRkxSIiwKICAiOCI6ICJQdWxsLVVwc1xuQXVmIFplaXQ6IDE1IC0gMTIgLSA5IFB1c' \
              b'2gtVXBzLFxuUnVzc2lhbiBTd2luZ3MsXG5TaXQtVXBzIiwKICAiMTEiOiAiNjAgU2VrdW5kZW4gQU1SQVA6IEJ1cnBlZX' \
              b'NcbkF1ZiBaZWl0OlxuNDAwIFJvcGUgSnVtcHNcbkludGVydmFsbGUgRkxSIiwKICAiMTMiOiAiRU1PVE0gZlx1MDBmY3' \
              b'IgNSBNaW51dGVuOiAxMiBBbWVyaWNhbiBTd2luZ3NcbjcgTWludXRlbiBBTVJBUDpcbjUgUHVzaC1VcHNcbjEwIFNpdC1' \
              b'VcHNcbjE1IEFpci1TcXVhdHMiLAogICIxNSI6ICI0eCBtYXguIFB1bGwtVXBzXG5BdWYgWmVpdDogMTAwMCBSb3BlI' \
              b'Ep1bXBzIHVuZCAxMDAgUHVzaC1VcHNcbjIgTWludXRlbiBIb2xsb3cgSG9sZCIsCiAgIjE4IjogIkVNT1RNIGZcdTAw' \
              b'ZmNyIDggTWludXRlbjpcbmdlcmFkZTogMTIgUnVzc2lhbiBTd2luZ3NcbnVuZ2VyYWRlOiAxMCBBbWVyaWNhbiBTd2luZ3' \
              b'NcblxuOCBSdW5kZW4gVGFiYXRhIEFpciBTcXVhdHMiLAogICIyMCI6ICI4IFJ1bmRlbiBUYWJhdGEgUm9wZSBKdW1wc1xuOC' \
              b'B4IDEwIG0gT25lIEFybSBPdmVyaGVhZCBMdW5nZXMiLAogICIyMiI6ICI0eCBtYXguIFB1c2gtdXBzXG5FTU9UTSBmXHUwMGZj' \
              b'ciAxMiBNaW51dGVuOlxuZ2VyYWRlOiAxMiBHb2JsZXQgU3F1YXRzXG51bmdlcmFkZTogMTAgUnVzc2lhbiBTd2luZ3NcblxuNCBS' \
              b'dW5kZW4gRkxSL01vdW50YWluIENsaW1iZXIgRmluaXNoZXIvRkxSIiwKICAiMjUiOiAiRU1PVE0gZlx1MDBmY3IgNSBNaW51dG' \
              b'VuOiBQdWxsLXVwc1xuOCBSdW5kZW4gVGFiYXRhIEJ1cnBlZXNcbjEyMCBTZWt1bmRlbiBMLUhvbGQiLAogICIyNyI6ICI0eCBtYXg' \
              b'uIFB1bGwtdXBzXG5cbkVNT1RNIGZcdTAwZmNyIDIwIE1pbnV0ZW46XG41IFB1c2gtdXBzLFxuNSBSdXNzaWFuIFN3aW5ncyxcbjUgU' \
              b'2l0LXVwcyIsCiAgIjI5IjogIjEwIE1pbnV0ZW4gUm9wZSBKdW1wc1xuRU1PVE0gZlx1MDBmY3IgMTAgTWludXRlbjpcbmdlcmFkZTo' \
              b'gMTAgR29ibGV0IFNxdWF0c1xudW5nZXJhZGU6IEhvbGxvdyBIb2xkIC9GTFIiLAogICIzMSI6ICJDaW5keVxuMTgwIFNla3VuZGVuI' \
              b'EwtU2l0LCBMLUhhbmcgdW5kIEhvbGxvdyBIb2xkIiwKICAiMzMiOiAiTHVuZ2VzLVZhcmlhdGlvbmVuLCBkYXp3aXNjaGVuIG1heCB' \
              b'QdWxsLXVwc1xuMkRlYXRoIGJ5IEJ1cnBlZXMiLAogICIzNSI6ICJBTVJBUCBmXHUwMGZjciA1IE1pbnV0ZW46XG42IFNESFAgJiA2' \
              b'IEJveCBKdW1wc1xuQXVmIFplaXQ6IDIxIC0gMTUgLSA5IEFtZXJpY2FuIFN3aW5ncyAmIFB1c2gtdXBzIiwKICAiMzciOiAiNiB4I' \
              b'DEwIEtldHRsZWJlbGwgRnJvbnQgU3F1YXRzLFxuZGF6d2lzY2hlbiBKdW1waW5nIFB1bGwtdXBzXG5cbjIxIC0gMTUgLSA5OiBEb3V' \
              b'ibGUgVW5kZXJzICYgQnVycGVlcyBvZGVyIFNpbmdsZXNcbnVuZCBCdXJwZWVzIGltIFdlY2hzZWxcblxuNCBSdW5kZW46IEZMUi9N' \
              b'b3VudGFpbiBDbGltYmVycy9GTFIiLAogICIzOSI6ICI0eCBtYXguIFB1bGwtdXBzXG5BTVJBUCBmXHUwMGZjciAxMCBNaW51dGVuO' \
              b'lxuNSBQdXNoLXVwc1xuMTAgQW1lcmljYW4gU3dpbmdzXG4xNSBTaXQtdXBzXG4iLAogICI0MSI6ICI1MDAwLW0tTGF1ZlxuMyB4ID' \
              b'EwIEhvbGxvdyBSb2NrLFxuZGF6d2lzY2hlbiAyIHggNDUgU2VrdW5kZW4gU3VwZXJtYW5cbiIsCiAgIjQzIjogIjIgUnVuZGVuOiB' \
              b'Nb3VudGFpbiBDbGltYmVycy9GTFIvTW91bnRhaW5cbkNsaW1iZXJzL0ZMUlxuQXVmIFplaXQ6IDEwMCBCdXJwZWVzXG4iLAogICI0N' \
              b'SI6ICIxMCBNaW51dGVuIFJvcGUgSnVtcDtcbjMgeCAxMiBIb2xsb3cgUm9ja1xuNSBSdW5kZW46XG5tYXguIFB1c2gtdXBzXG4xMCB' \
              b'Cb3ggSnVtcHNcbjUwIFNpbmdsZXMgdW5icm9rZW5cbiIsCiAgIjQ3IjogIjMgRHVyY2hnXHUwMGU0bmdlIFB1bGwtdXBzXG5FTU9UT' \
              b'SBmXHUwMGZjciA4IE1pbnV0ZW46IEFtZXJpY2FuIFN3aW5nc1xuRU1PVE0gZlx1MDBmY3IgOCBNaW51dGVuOiBIb2xsb3cgUm9ja1x' \
              b'uRU1PVE0gZlx1MDBmY3IgOCBNaW51dGVuOiBHb2JsZXQgTHVuZ2VzXG4iLAogICI0OSI6ICIxMCBNaW51dGVuIFJvcGUgSnVtcFxuN' \
              b'yBNaW51dGVuIEFNUkFQOlxuNSBQdXNoLXVwc1xuMTAgU2l0LXVwc1xuMTUgQWlyIFNxdWF0c1xuIiwKICAiNTEiOiAiNHggbWF4LiB' \
              b'QdWxsLXVwc1xuNSBSdW5kZW46XG42MCBTZWt1bmRlbiBBTVJBUDpcbjMgQnVycGVlc1xuNSBBbWVyaWNhbiBTd2luZ3NcbiIsCiAg' \
              b'IjUzIjogIjUgTWludXRlbiBSb3BlIEp1bXBzXG5FTU9UTSBmXHUwMGZjciAzMCBNaW51dGVuOlxuNSBQdXNoLXVwc1xuNSBSdXNza' \
              b'WFuIFN3aW5nc1xuNSBTaXQtdXBzXG5KZWRlIDUuIE1pbnV0ZTogMTAgQnVycGVlc1xuIiwKICAiNTUiOiAiNSBNaW51dGVuIFJvcG' \
              b'UgSnVtcHNcbjIgTWludXRlbiBBTVJBUDogRG91YmxlIFVuZGVyc1xuNHggbWF4LiBIb2xsb3cgSG9sZFxuOCBSdW5' \
              b'kZW4gVGFiYXRhIEJ1cnBlZXMiCn0='

        ret = base64.b64decode(ret_encoded).decode(encoding="utf-8")
        ret = json.loads(ret)
        ret = {int(i): j for i, j in ret.items()}

        return ret

    def create_workouts_from_input(self, input):
        workouts = []
        for day, workout in input.items():
            workouts.append(
                CalendarEvent(f"Workout Day {day}",
                              self._calculate_workout_date(day),
                              workout))

        return workouts

    def _calculate_workout_date(self, sections_index):
        return self.workout_start + (timedelta(days=1) * sections_index)


if __name__ == "__main__":
    cross_fit = CrossFit()
    cross_fit.build()
