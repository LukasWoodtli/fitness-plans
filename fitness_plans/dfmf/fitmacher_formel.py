#!/usr/bin/env python3
# -*- coding: utf-8 -*-
import base64
import pathlib
from datetime import date, timedelta
import re
import os

from fitness_plans.lib.abstract_fitness_plan import FitnessPlan
from fitness_plans.lib.workout_calendar import CalendarEvent, WorkoutCalendar

SCRIPT_DIR = os.path.dirname(os.path.realpath(__file__))


class FitMacherFormel(FitnessPlan):
    INPUT_FILE = os.path.join(SCRIPT_DIR, 'dfmf.encoded.txt')
    CALENDAR_NAME = "DieFitMacherFormel"

    def __init__(self, workout_start_date=date.today(), first_workout=1, output_dir=None):
        super().__init__()
        self.workout_start = workout_start_date
        self.first_workout = first_workout
        self.output_dir = output_dir

    def read_input_file(self):
        with open(self.INPUT_FILE, 'rb') as infile:
            encoded_data = infile.read()
            data = base64.decodebytes(encoded_data)
            data = data.decode('UTF-8')
            return data

    def create_workouts_from_input(self, input):
        workouts = self._split_workouts(self.read_input_file())
        return workouts

    def filter_workouts(self, workouts):
        return [workout for workout in workouts if int(workout.summary.split(".")[0]) >= self.first_workout]

    def create_workout_calendar_ics(self, workouts):
        cal = WorkoutCalendar(workouts)
        workouts_as_ical = cal.get_all_as_ical()
        return workouts_as_ical

    def save_calendar(self, ics):
        output_file_path = f'{self.CALENDAR_NAME}.ics'
        if self.output_dir:
            pathlib.Path(self.output_dir).mkdir(parents=True, exist_ok=True)
            output_file_path = os.path.join(self.output_dir, output_file_path)
        with open(output_file_path, 'w') as f_out:
            f_out.writelines(ics)

    def _split_workouts(self, data):
        split_day_re = r"([0-9]{1,2}\. TAG[^\n]*)"

        all_workouts = []
        sections = re.split(split_day_re, data, flags=re.M)
        sections = sections[1:]  # first entry is empty
        assert len(sections) % 2 == 0
        for s in range(0, int(len(sections)/2)):
            event = self._create_workout_event(s, sections)
            all_workouts.append(event)
        return all_workouts

    def _create_workout_event(self, sections_index, sections):
        title_index = sections_index * 2
        title = sections[title_index]
        description = sections[title_index + 1]
        description = self._create_description(description, title)
        workout_date = self._calculate_workout_date(sections_index)
        event = CalendarEvent(title, workout_date, description)
        return event

    @staticmethod
    def _create_description(description, title):
        description = re.sub("\n\n+", "\n\n", description)
        description = title + description
        return description

    def _calculate_workout_date(self, sections_index):
        return self.workout_start + (timedelta(days=1) * sections_index)


if __name__ == "__main__":
    fitmacher_formel = FitMacherFormel()
    fitmacher_formel.build()
