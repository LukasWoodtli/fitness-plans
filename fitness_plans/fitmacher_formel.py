#!/usr/bin/env python3
# -*- coding: utf-8 -*-

from datetime import date, timedelta
import re
import os

from fitness_plans.workout_calendar import CalendarEvent, WorkoutCalendar

SCRIPT_DIR = os.path.dirname(os.path.realpath(__file__))


class FitMacherFormel:
    INPUT_FILE = os.path.join(SCRIPT_DIR, '../dfmf.txt')
    CALENDAR_NAME = "DieFitMacherFormel"

    def __init__(self, workout_start_date=date.today(), first_workout=1):
        self.workout_start = workout_start_date
        self.first_workout = first_workout

    def create_workout_calendar(self):
        with open(f'{self.CALENDAR_NAME}.ics', 'w') as f_out:
            workouts_as_ical = self._create_workouts_from_text()
            f_out.writelines(workouts_as_ical)

    def _create_workouts_from_text(self):
        workouts = self._split_workouts(self._read_input_file())
        workouts = self._filter_skipped_workouts(workouts)
        cal = WorkoutCalendar(workouts)
        return cal.get_all_as_ical()

    def _read_input_file(self):
        with open(self.INPUT_FILE, 'r', encoding='utf-8') as infile:
            data = infile.read()
            data = self._strip_unneeded_text(data)
            return data

    @staticmethod
    def _strip_unneeded_text(data):
        data = FitMacherFormel._strip_unneeded_begin_of_text(data)
        data = FitMacherFormel._strip_unneeded_end_of_text(data)
        return data

    @staticmethod
    def _strip_unneeded_end_of_text(data):
        end_text = "RIM TIM MUART NENIEM EBEL"  # just a bit obfuscated
        end_text = end_text[::-1]
        data = data.split(end_text)[0]  # remove end
        return data

    @staticmethod
    def _strip_unneeded_begin_of_text(data):
        begin_text = "GAT .1"  # just a bit obfuscated
        begin_text = begin_text[::-1]
        data = data[data.find(begin_text):]  # remove begin
        return data

    def _filter_skipped_workouts(self, workouts):
        return [workout for workout in workouts if int(workout.summary.split(".")[0]) >= self.first_workout]

    def _split_workouts(self, data):
        split_day_re = r"([0-9]{1,2}\. TAG[^\n]*)"

        all_workouts = []
        sections = re.split(split_day_re, data, flags=re.M)
        sections = sections[1:]  # first entry is empty
        assert len(sections) % 2 == 0
        for s in range(0, int(len(sections)/2)):
            event = self._create_workout_event(s, sections)
            all_workouts.append(event)
        return all_workouts

    def _create_workout_event(self, sections_index, sections):
        title_index = sections_index * 2
        title = sections[title_index]
        description = sections[title_index + 1]
        description = self._create_description(description, title)
        workout_date = self._calculate_workout_date(sections_index)
        event = CalendarEvent(title, workout_date, description)
        return event

    @staticmethod
    def _create_description(description, title):
        description = re.sub("\n\n+", "\n\n", description)
        description = title + description
        return description

    def _calculate_workout_date(self, sections_index):
        return self.workout_start + (timedelta(days=1) * sections_index)


if __name__ == "__main__":
    fitmacher_formel = FitMacherFormel()
    fitmacher_formel.create_workout_calendar()
