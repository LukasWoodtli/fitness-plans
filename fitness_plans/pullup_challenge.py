#!/usr/bin/env python3
from datetime import timedelta, date, datetime
import requests
from bs4 import BeautifulSoup

from fitness_plans.workout_calendar import CalendarEvent, WorkoutCalendar


class Workout:
    def __init__(self, week, title, subtitle):
        self.week = week
        self.title = title
        self.subtitle = subtitle
        self.easy_set = []
        self.medium_set = []
        self.hard_set = []

    def get_full_title(self):
        return f"{self.title} (Week {self.week})"

    def _get_workout_set_text(self, level):
        text = f"{level.title()}:\n"

        if level.lower() == "easy":
            workout_set = self.easy_set
        elif level.lower() == "medium":
            workout_set = self.medium_set
        elif level.lower() == "hard":
            workout_set = self.hard_set
        else:
            raise Exception(f"Level not known: {level}")

        prerequisites = workout_set[0]
        text += f"Prerequisites: {prerequisites}\n\n"

        for i, s in enumerate(workout_set[1:], 1):
            text += f"\tSet {i}: {s}\n"
        text += "\n\n"

        return text

    def get_description_text(self):
        text = f"{self.subtitle}\n\n"
        text += self._get_workout_set_text("Easy")
        #text += self._get_workout_set_text("Medium")
        #text += self._get_workout_set_text("Hard")
        return text

    def __repr__(self):
        text = self.get_full_title() + "\n"
        text += self.get_description_text() + "\n"

        return text

    def append_workout_step(self, easy_workout, medium_workout, hard_workout):
        self.easy_set.append(easy_workout)
        self.medium_set.append(medium_workout)
        self.hard_set.append(hard_workout)


class PullUpChallenge:

    def __init__(self, week_no=1, start_date=date.today(), workout_frequecy=timedelta(days=2)):
        self._week_no = 1
        self.workout_start = start_date
        self.workout_frequency = workout_frequecy

    def _get_all_workouts(self):
        table_body = self._fetch_page_and_parse_table()

        rows = table_body.find_all('tr')
        all_workouts = []
        for row in rows:
            cols = row.find_all('td')
            if not cols:
                text = row.text
                text = text.splitlines()
                text = [line.strip() for line in text if line.strip()]
                assert len(text) == 2
                title = text[0]
                assert title.startswith("Day")
                subtitle = text[1]
                assert subtitle.startswith("Rest")
                workout_day = Workout(self._week_no, title, subtitle)
                all_workouts.append(workout_day)

            else:
                assert workout_day
                cols = [ele.text.strip() for ele in cols]
                assert len(cols) == 3
                workout_day.append_workout_step(*cols)

        return all_workouts

    def _fetch_page_and_parse_table(self):
        page = self._fetch_page()
        return self._parse_table(page)

    def _fetch_page(self):
        request = requests.get('https://twentypullups.com/')
        soup = BeautifulSoup(request.text, 'html.parser')
        return soup

    def _parse_table(self, html_soup):
        week_no_text = f"Week {self._week_no}"
        # link to week
        weeks = [i.get('href') for i in html_soup.find_all('a') if week_no_text in i.text]

        # parse table
        week_page_request = requests.get(weeks[0])
        week_page = BeautifulSoup(week_page_request.text, 'html.parser')

        table = week_page.find('table')
        table_body = table.find('tbody')

        return table_body

    def _workouts_to_cal_events(self):
        workouts = self._get_all_workouts()
        cal_events = []
        workout_date = self.workout_start
        for workout in workouts:
            title = workout.get_full_title()
            description = workout.get_description_text()
            event = CalendarEvent(title, workout_date, description)
            workout_date = workout_date + self.workout_frequency
            cal_events.append(event)
        return cal_events

    def _create_workouts_from_text(self):
        events = self._workouts_to_cal_events()
        cal = WorkoutCalendar(events)
        return cal.get_all_as_ical()

    def create_workout_calendar(self):
        cal_text = self._create_workouts_from_text()
        with open("pullup.ics", 'w') as out_file:
            out_file.writelines(cal_text)


if __name__ == "__main__":
    challenge = PullUpChallenge()
    challenge.create_workout_calendar()