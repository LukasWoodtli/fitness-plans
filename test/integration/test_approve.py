import os
import shutil
from datetime import date
from unittest.mock import patch

from approvaltests import Options, verify, verify_file
from approvaltests.scrubbers import create_regex_scrubber

from fitness_plans.crossfit.crossfit import CrossFit
from fitness_plans.dfmf.fitmacher_formel import FitMacherFormel
from fitness_plans.pullup_challenge import PullUpChallenge
from fitness_plans.push_ups import PushUps

OPTIONS_WITH_SCRUBBER = Options()\
    .for_file.with_extension(".txt")\
    .with_scrubber(create_regex_scrubber("DTSTAMP;VALUE=DATE-TIME.*", "[creation date]"))


@patch('fitness_plans.dfmf.fitmacher_formel.FitMacherFormel.save_calendar')
def test_fitmacher_formel(mock_save_calendar):
    fmf = FitMacherFormel(date(2020, 5, 29))
    fmf.build()
    cal = mock_save_calendar.call_args.args[0]
    verify(cal, options=OPTIONS_WITH_SCRUBBER)


def test_fitmacher_formel_output_file():
    test_output_dir = os.path.join(os.path.dirname(__file__), 'test_out')
    fmf = FitMacherFormel(date(2021, 10, 11), output_dir=test_output_dir)
    fmf.build()
    out_file = os.path.join(test_output_dir, "DieFitMacherFormel.ics")

    verify_file(out_file, options=OPTIONS_WITH_SCRUBBER)

def test_cross_fit_output_file():
    test_output_dir = os.path.join(os.path.dirname(__file__), 'test_out')
    cross_fit = CrossFit(date(2022, 12, 11), output_dir=test_output_dir)
    cross_fit.build()
    out_file = os.path.join(test_output_dir, "Cross Fit.ics")

    verify_file(out_file, options=OPTIONS_WITH_SCRUBBER)

def test_pullup_challenge():
    challenge = PullUpChallenge(start_date=date(2020, 5, 30))
    calendar = challenge._create_workouts_from_text()
    verify(calendar, options=OPTIONS_WITH_SCRUBBER)


def test_pushups():
    push_ups = PushUps(start_date=date(2020, 5, 30))
    calendar = push_ups._create_workouts()
    verify(calendar, options=OPTIONS_WITH_SCRUBBER)


@patch('fitness_plans.dfmf.fitmacher_formel.FitMacherFormel.save_calendar')
@patch('fitness_plans.dfmf.fitmacher_formel.FitMacherFormel.get_input')
def test_create_workout_calendar(mock_read_input_file, mock_save_calendar):
    mock_read_input_file.return_value = \
        "1. TAG: Hello 1\n" \
        "Workout 1\n" \
        "2. TAG: Hello 2\n" \
        "Workout 2\n" \
        "3. TAG: Hello 3\n" \
        "Workout 3"
    fmf = FitMacherFormel(workout_start_date=date(2021, 10, 11))
    fmf.build()
    calendar = mock_save_calendar.call_args.args[0]
    verify(calendar, options=OPTIONS_WITH_SCRUBBER)
