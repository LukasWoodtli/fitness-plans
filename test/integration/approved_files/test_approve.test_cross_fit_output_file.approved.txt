BEGIN:VCALENDAR
VERSION:2.0
PRODID:Fitness
BEGIN:VEVENT
SUMMARY:Workout Day 1
DTSTART;VALUE=DATE:20221212
DTEND;VALUE=DATE:20221212
[creation date]
DESCRIPTION:EMOTM für 5 Minuten: 15 Air Squats\nAuf Zeit: 10...1 Push-Up 
 und Sit-Up
END:VEVENT
BEGIN:VEVENT
SUMMARY:Workout Day 4
DTSTART;VALUE=DATE:20221215
DTEND;VALUE=DATE:20221215
[creation date]
DESCRIPTION:EMOTM für 5 Minuten:\n15 Russian Swings\nAuf Zeit: 250 Rope J
 umps
END:VEVENT
BEGIN:VEVENT
SUMMARY:Workout Day 6
DTSTART;VALUE=DATE:20221217
DTEND;VALUE=DATE:20221217
[creation date]
DESCRIPTION:10...1 Goblet Squats & Push-Ups\nIntervalle FLR
END:VEVENT
BEGIN:VEVENT
SUMMARY:Workout Day 8
DTSTART;VALUE=DATE:20221219
DTEND;VALUE=DATE:20221219
[creation date]
DESCRIPTION:Pull-Ups\nAuf Zeit: 15 - 12 - 9 Push-Ups\,\nRussian Swings\,\n
 Sit-Ups
END:VEVENT
BEGIN:VEVENT
SUMMARY:Workout Day 11
DTSTART;VALUE=DATE:20221222
DTEND;VALUE=DATE:20221222
[creation date]
DESCRIPTION:60 Sekunden AMRAP: Burpees\nAuf Zeit:\n400 Rope Jumps\nInterva
 lle FLR
END:VEVENT
BEGIN:VEVENT
SUMMARY:Workout Day 13
DTSTART;VALUE=DATE:20221224
DTEND;VALUE=DATE:20221224
[creation date]
DESCRIPTION:EMOTM für 5 Minuten: 12 American Swings\n7 Minuten AMRAP:\n5 
 Push-Ups\n10 Sit-Ups\n15 Air-Squats
END:VEVENT
BEGIN:VEVENT
SUMMARY:Workout Day 15
DTSTART;VALUE=DATE:20221226
DTEND;VALUE=DATE:20221226
[creation date]
DESCRIPTION:4x max. Pull-Ups\nAuf Zeit: 1000 Rope Jumps und 100 Push-Ups\n
 2 Minuten Hollow Hold
END:VEVENT
BEGIN:VEVENT
SUMMARY:Workout Day 18
DTSTART;VALUE=DATE:20221229
DTEND;VALUE=DATE:20221229
[creation date]
DESCRIPTION:EMOTM für 8 Minuten:\ngerade: 12 Russian Swings\nungerade: 10
  American Swings\n\n8 Runden Tabata Air Squats
END:VEVENT
BEGIN:VEVENT
SUMMARY:Workout Day 20
DTSTART;VALUE=DATE:20221231
DTEND;VALUE=DATE:20221231
[creation date]
DESCRIPTION:8 Runden Tabata Rope Jumps\n8 x 10 m One Arm Overhead Lunges
END:VEVENT
BEGIN:VEVENT
SUMMARY:Workout Day 22
DTSTART;VALUE=DATE:20230102
DTEND;VALUE=DATE:20230102
[creation date]
DESCRIPTION:4x max. Push-ups\nEMOTM für 12 Minuten:\ngerade: 12 Goblet Sq
 uats\nungerade: 10 Russian Swings\n\n4 Runden FLR/Mountain Climber Finishe
 r/FLR
END:VEVENT
BEGIN:VEVENT
SUMMARY:Workout Day 25
DTSTART;VALUE=DATE:20230105
DTEND;VALUE=DATE:20230105
[creation date]
DESCRIPTION:EMOTM für 5 Minuten: Pull-ups\n8 Runden Tabata Burpees\n120 S
 ekunden L-Hold
END:VEVENT
BEGIN:VEVENT
SUMMARY:Workout Day 27
DTSTART;VALUE=DATE:20230107
DTEND;VALUE=DATE:20230107
[creation date]
DESCRIPTION:4x max. Pull-ups\n\nEMOTM für 20 Minuten:\n5 Push-ups\,\n5 Ru
 ssian Swings\,\n5 Sit-ups
END:VEVENT
BEGIN:VEVENT
SUMMARY:Workout Day 29
DTSTART;VALUE=DATE:20230109
DTEND;VALUE=DATE:20230109
[creation date]
DESCRIPTION:10 Minuten Rope Jumps\nEMOTM für 10 Minuten:\ngerade: 10 Gobl
 et Squats\nungerade: Hollow Hold /FLR
END:VEVENT
BEGIN:VEVENT
SUMMARY:Workout Day 31
DTSTART;VALUE=DATE:20230111
DTEND;VALUE=DATE:20230111
[creation date]
DESCRIPTION:Cindy\n180 Sekunden L-Sit\, L-Hang und Hollow Hold
END:VEVENT
BEGIN:VEVENT
SUMMARY:Workout Day 33
DTSTART;VALUE=DATE:20230113
DTEND;VALUE=DATE:20230113
[creation date]
DESCRIPTION:Lunges-Variationen\, dazwischen max Pull-ups\n2Death by Burpee
 s
END:VEVENT
BEGIN:VEVENT
SUMMARY:Workout Day 35
DTSTART;VALUE=DATE:20230115
DTEND;VALUE=DATE:20230115
[creation date]
DESCRIPTION:AMRAP für 5 Minuten:\n6 SDHP & 6 Box Jumps\nAuf Zeit: 21 - 15
  - 9 American Swings & Push-ups
END:VEVENT
BEGIN:VEVENT
SUMMARY:Workout Day 37
DTSTART;VALUE=DATE:20230117
DTEND;VALUE=DATE:20230117
[creation date]
DESCRIPTION:6 x 10 Kettlebell Front Squats\,\ndazwischen Jumping Pull-ups\
 n\n21 - 15 - 9: Double Unders & Burpees oder Singles\nund Burpees im Wechs
 el\n\n4 Runden: FLR/Mountain Climbers/FLR
END:VEVENT
BEGIN:VEVENT
SUMMARY:Workout Day 39
DTSTART;VALUE=DATE:20230119
DTEND;VALUE=DATE:20230119
[creation date]
DESCRIPTION:4x max. Pull-ups\nAMRAP für 10 Minuten:\n5 Push-ups\n10 Ameri
 can Swings\n15 Sit-ups\n
END:VEVENT
BEGIN:VEVENT
SUMMARY:Workout Day 41
DTSTART;VALUE=DATE:20230121
DTEND;VALUE=DATE:20230121
[creation date]
DESCRIPTION:5000-m-Lauf\n3 x 10 Hollow Rock\,\ndazwischen 2 x 45 Sekunden 
 Superman\n
END:VEVENT
BEGIN:VEVENT
SUMMARY:Workout Day 43
DTSTART;VALUE=DATE:20230123
DTEND;VALUE=DATE:20230123
[creation date]
DESCRIPTION:2 Runden: Mountain Climbers/FLR/Mountain\nClimbers/FLR\nAuf Ze
 it: 100 Burpees\n
END:VEVENT
BEGIN:VEVENT
SUMMARY:Workout Day 45
DTSTART;VALUE=DATE:20230125
DTEND;VALUE=DATE:20230125
[creation date]
DESCRIPTION:10 Minuten Rope Jump\;\n3 x 12 Hollow Rock\n5 Runden:\nmax. Pu
 sh-ups\n10 Box Jumps\n50 Singles unbroken\n
END:VEVENT
BEGIN:VEVENT
SUMMARY:Workout Day 47
DTSTART;VALUE=DATE:20230127
DTEND;VALUE=DATE:20230127
[creation date]
DESCRIPTION:3 Durchgänge Pull-ups\nEMOTM für 8 Minuten: American Swings\
 nEMOTM für 8 Minuten: Hollow Rock\nEMOTM für 8 Minuten: Goblet Lunges\n
END:VEVENT
BEGIN:VEVENT
SUMMARY:Workout Day 49
DTSTART;VALUE=DATE:20230129
DTEND;VALUE=DATE:20230129
[creation date]
DESCRIPTION:10 Minuten Rope Jump\n7 Minuten AMRAP:\n5 Push-ups\n10 Sit-ups
 \n15 Air Squats\n
END:VEVENT
BEGIN:VEVENT
SUMMARY:Workout Day 51
DTSTART;VALUE=DATE:20230131
DTEND;VALUE=DATE:20230131
[creation date]
DESCRIPTION:4x max. Pull-ups\n5 Runden:\n60 Sekunden AMRAP:\n3 Burpees\n5 
 American Swings\n
END:VEVENT
BEGIN:VEVENT
SUMMARY:Workout Day 53
DTSTART;VALUE=DATE:20230202
DTEND;VALUE=DATE:20230202
[creation date]
DESCRIPTION:5 Minuten Rope Jumps\nEMOTM für 30 Minuten:\n5 Push-ups\n5 Ru
 ssian Swings\n5 Sit-ups\nJede 5. Minute: 10 Burpees\n
END:VEVENT
BEGIN:VEVENT
SUMMARY:Workout Day 55
DTSTART;VALUE=DATE:20230204
DTEND;VALUE=DATE:20230204
[creation date]
DESCRIPTION:5 Minuten Rope Jumps\n2 Minuten AMRAP: Double Unders\n4x max. 
 Hollow Hold\n8 Runden Tabata Burpees
END:VEVENT
END:VCALENDAR
